### PR TITLE
[MM-29806] Add end-2-end tests to bot list command

### PR DESCRIPTION
**Pull request by lucianomagrao**
#### Summary

Add e2e tests to `mmctl bot list` command.

#### Ticket Link

Fixes: https://github.com/mattermost/mattermost-server/issues/16002
JIRA: https://mattermost.atlassian.net/browse/MM-29806

### CODE REVIEW

**Comment by hanzei** [comment]
Thanks for the PR @lucianomagrao :+1: Heads up that CI reported a couple of linter errors. Could you please take a look at them? You can find the logs [here](https://app.circleci.com/pipelines/github/mattermost/mmctl/2336/workflows/8d0087c0-07ff-4743-9d1a-b701cf8682bb/jobs/8417/parallel-runs/0/steps/0-106).

File: commands/bot_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
```
Comment (agnivade) [inline commands/bot_e2e_test.go:11]: Let's make this to be the first group.

File: commands/bot_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		deletedBot, appErr = s.th.App.UpdateBotActive(deletedBot.UserId, false)
+		s.Require().Nil(appErr)
+
+		cmd := &cobra.Command{}
+		cmd.Flags().Bool("all", true, "")
+
+		err := botListCmdF(c, cmd, []string{})
+		s.Require().Nil(err)
+		s.Require().Equal(3, len(printer.GetLines()))
```
Comment (agnivade) [inline commands/bot_e2e_test.go:140]: Let's also match the bot names.

**Review by agnivade — COMMENTED** [review]
Looking good! Just some small changes needed.

**Review by agnivade — APPROVED** [review]

File: commands/bot_e2e_test.go [hunk 3]
```diff
@@ -4,13 +4,169 @@
...
+
+	s.Run("List Bots without permission", func() {
+		printer.Clean()
+
+		bot, appErr := s.th.App.CreateBot(&model.Bot{Username: model.NewId(), OwnerId: s.th.BasicUser.Id})
+		s.Require().Nil(appErr)
+		defer func() {
+			err := s.th.App.PermanentDeleteBot(bot.UserId)
+			s.Require().Nil(err)
+		}()
```
Comment (streamer45) [inline commands/bot_e2e_test.go:160]: I think can remove this since we are purely testing for permissions.

**Review by streamer45 — COMMENTED** [review]
@lucianomagrao Thank you for your contribution :+1: PR looks good.
I've left a couple of suggestions.

**Review by streamer45 — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @lucianomagrao @streamer45 - looks good to me 👍

**Comment by hanzei** [comment]
Thanks @lucianomagrao :+1:

**Pull request merged**